ast/src/analyzed/visitor.rs

```diff
@@ -40,4 +40,13 @@ impl<T> AllChildren<AlgebraicExpression<T>> for AlgebraicExpression<T> {
     fn all_children(&self) -> Box<dyn Iterator<Item = &AlgebraicExpression<T>> + '_> {
         Box::new(iter::once(self).chain(self.children().flat_map(|e| e.all_children())))
     }
+
+    fn all_children_mut(&mut self) -> Box<dyn Iterator<Item = &mut AlgebraicExpression<T>> + '_> {
```

Comment:
This PR is already huge, please try to factor out this change (and mabye other more or less unrelated ones)